### PR TITLE
WASM corlib support

### PR DESCRIPTION
**Pull request by kumpera**
This PR lands an initial set of fixes to deal with corlib failures.

It disables a lot of tests that requires multiple threads plus make the ThreadPool and finalizers work.

### CODE REVIEW

File: sdks/wasm/Makefile [hunk 1]
```diff
@@ -75,18 +75,24 @@ managed/nunitlite.dll: $(TOP)/mcs/class/lib/wasm/nunitlite.dll | managed/
 	cd $(TOP)/mcs/tools/nunit-lite && make PROFILE=wasm
 	cp $< $@
 
+<<<<<<< HEAD
```
Comment (akoeplinger) [inline sdks/wasm/Makefile]: remove conflict marker

File: mcs/class/corlib/Test/System.Threading/ReaderWriterLockTest.cs [hunk 2]
```diff
@@ -94,6 +94,7 @@ private void IsWriterLockHeld_2 ()
 		}
 				
 		[Test]
+		[Category ("MultiThreaded")]
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System.Threading/ReaderWriterLockTest.cs]: I don't see this category being disabled anywhere in sdks/ ?

File: sdks/wasm/Makefile (hunk 1 above)
Comment (kumpera) [reply sdks/wasm/Makefile]: uggg, I had stashed some random changes that unfortunately has the fix for it :/

File: mcs/class/corlib/Test/System.Threading/ReaderWriterLockTest.cs (hunk 2 above)
Comment (kumpera) [reply mcs/class/corlib/Test/System.Threading/ReaderWriterLockTest.cs]: Yeah, that's missing from this PR, I got some fixing to do here anyways.

**Comment by kumpera** [comment]
Rebased this PR, it fixes the issues mentioned by @akoeplinger plus the merge conflict.

**Pull request merged**